### PR TITLE
[REEF-1231] Too many open files in NetworkConnectionServiceTest

### PR DESCRIPTION
**Pull request by taegeonum**
This pull request addressed the issue [REEF-1231](https://issues.apache.org/jira/browse/REEF-1231) by
* adding a public constructor of `NameLookupClient`
* removing `NameLookupClient` injection from `NameClient`

JIRA:
[REEF-1231] https://issues.apache.org/jira/browse/REEF-1231

Pull Request:
Closes #

### CODE REVIEW

**Comment by bgchun** [comment]
@taegeonum Thanks. CI's not completed yet. I will take a look.

File: lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameClient.java [hunk 1]
```diff
@@ -86,7 +85,8 @@ private NameClient(
             new NamingResponseHandler(replyLookupQueue, replyRegisterQueue), codec)),
         null, retryCount, retryTimeout);
 
-    this.lookupClient = lookupClient;
+    this.lookupClient = new NameLookupClient(serverAddr, serverPort, timeout, factory,
+        retryCount, retryTimeout, replyLookupQueue, transport);
```
Comment (bgchun) [inline lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameClient.java]: `transport` and `this.transport`: inconsistent use. 
Could you following the reef convention? We have discussed about it.

**Comment by bgchun** [comment]
@taegeonum I left a minor comment. Can you resolve it?

**Comment by taegeonum** [comment]
@bgchun I've addressed your comment. Thanks!

File: lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameLookupClient.java [hunk 2]
```diff
@@ -72,6 +72,37 @@
...
+   * @param serverAddr a server address
+   * @param serverPort a server port number
+   * @param timeout    request timeout in ms
+   * @param factory    an identifier factory
+   * @param retryCount a count of retrying lookup
+   * @param retryTimeout retry timeout
+   * @param replyQueue a reply queue
+   * @param transport  a transport
+   */
+  public NameLookupClient(final String serverAddr,
```
Comment (markusweimer) [inline lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameLookupClient.java]: Does this have to be `public`?

**Comment by bgchun** [comment]
@taegeonum Can you address Markus's comment?

File: lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameLookupClient.java (hunk 2 above)
Comment (taegeonum) [inline lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameLookupClient.java]: No. I've remove `public`. Thanks!

**Comment by taegeonum** [comment]
@bgchun I've addressed Markus's comment. Thanks.

**Comment by markusweimer** [comment]
This now LGTM. I will test and merge.

**Pull request closed without merging**